frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.html

```diff
@@ -0,0 +1,237 @@
+                        <div *ngIf="work.controls.dateEnd.invalid && work.controls.dateEnd.touched" class="error-message">
+                            <div *ngIf="work.controls.dateEnd.errors.required">
+                                Finish date is required.
+                            </div>
+                            <div *ngIf="work.controls.dateEnd.errors.pattern">
+                                Invalid Finish date!
+                            </div>
+                        </div>
+                    </td>
+                    <td>
+                        <div class="btn btn-light pull-right" (click)="remove(i, 'workCandidateList')">
+                            <img class="far fa-trash-alt align-middle" />
+                        </div>
+                    </td>
+                </tr>
+            </tbody>
+        </table>
+        <div class="addNews">
+            <button (click)="additionWork()">
+                Add work
+            </button>
+        </div>
+        <div>
+            <h3>
+                Choose discipline
+            </h3>
+            <div formArrayName="disciplineList">
+                <div class='selectDisciplines' *ngFor="let discipline of candidateForm.controls.disciplineList.controls; let i = index" [formGroupName]="i">
+                    <select id="inputState" class="form-control " formControlName="id" value="1">
+                        <option value="" hidden disabled selected>Choose...</option>
+                        <option *ngFor="let dis of disciplines" value="{{dis.id}}" selected>{{dis.name}}</option>
+                    </select>
+                    <div class="btn btn-light pull-right" (click)="remove(i,'disciplineList')" *ngIf="candidateForm.controls.disciplineList.controls.length > 1">
+                        <img class="far fa-trash-alt align-middle" />
+                    </div><br><br>
+                    <div *ngIf="discipline.controls.id.invalid && discipline.controls.id.touched" class="error-message">
+                        <div *ngIf="discipline.controls.id.errors.required">
+                            Discipline not selected!!!
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <div class="btn btn-light col-md-12" (click)="additionDiscipline()">
+                <a class="fas fa-plus align-middle"></a>
+                Assign to the additional discipline
+            </div>
+        </div>
+        <div class="text-center add-candidates">
+            <a class="btn btn-primary" (click)="createCandidate()">Add Candidate</a>
+            <a class="btn btn-secondary" (click)="change()">Close</a>
+            <br>
+            <br>
+            <span *ngIf="displayErrorMessage" class="error-message">Fill in all the fields</span>
+        </div>
+    </form>
+</div>
+<app-message-window *ngIf="displayMessage" [message]="viewMessage" (displayWindowMessage)="change()"></app-message-window>
```

Comment:
This template is too big and there is a lot of repeating code. Maybe it makes sense to create a special component.